vsintegration/src/FSharp.VS.FSI/fsiSessionToolWindow.fs

```diff
@@ -534,17 +534,16 @@ type internal FsiToolWindow() as this =
             executeTextNoHistory null text
         with _ -> ()
 
-    let executeInteraction dbgBreak dir (filename: string) topLine text =
-        // Preserving previous functionality, including the #directives...
-        let interaction =
-            "\n"
-          + (sprintf "# silentCd @\"%s\" ;; " dir) + "\n"
-          + (if dbgBreak then "# dbgbreak\n" else "")
-          + (sprintf "# %d @\"%s\" " topLine filename) + "\n"
-          + text + "\n"
-          + "# 1 \"stdin\"" + "\n" (* stdin line number reset code *)
-          + ";;" + "\n"
-
+    let executeInteraction dbgBreak dir filename topLine (text:string) =
+        let interaction = $"""
+#interactiveprompt "hide"
+#silentCd @"{dir}";;
+{if dbgBreak then "#dbgbreak" else ""}
+#{topLine} @"{filename}"
+{text.ToString()};;
```

Comment:
Why to call .ToString() on a string ?

```suggestion
{text};;
```

Reply:
Also unsure about the trailing ;; here, as those come later.. ?

Reply:
I believe text is actually a stringbuilder and so it needs to be ToString() if you look in the original it is concatenated with '+' not printed with sprintf "%s".  When I get to my desk tomorrow, I can check if you like.

As for the ';;' I think there was a scenario where it was necessary, but I can't for the life of me remember what it was.  So again, I can check it out when I get to my desk.

Thanks

Reply:
I believe `ToString` would be called anyway in an interpolated expression, but I am only 90% sure.

Reply:
Actually, I declared text a string.

However, without ToString()
![image](https://user-images.githubusercontent.com/5175830/204112289-c9cf0058-d1c6-4fb2-b1d0-d59b29db0f23.png)
With ToString():
![image](https://user-images.githubusercontent.com/5175830/204112329-5a0f3149-1e4c-4bae-bb0a-7234cca2c85c.png)

For the non-ToString() version the compiler produces this:
````
  FSharp.Compiler.Service -> C:\kevinransom\fsharp\artifacts\bin\FSharp.Compiler.Service\Debug\netstandard2.0\FSharp.Compiler.Service.d
  ll
C:\kevinransom\fsharp\vsintegration\src\FSharp.VS.FSI\fsiSessionToolWindow.fs(538,9): error FS0588: The block following this 'let' is u
nfinished. Every code block is an expression and must have a result. 'let' cannot be the final code element in a block. Consider giving
 this block an explicit result. [C:\kevinransom\fsharp\vsintegration\src\FSharp.VS.FSI\FSharp.VS.FSI.fsproj]
C:\kevinransom\fsharp\vsintegration\src\FSharp.VS.FSI\fsiSessionToolWindow.fs(547,9): error FS0010: Unexpected identifier in binding. E
xpected incomplete structured construct at or before this point or other token. [C:\kevinransom\fsharp\vsintegration\src\FSharp.VS.FSI\
FSharp.VS.FSI.fsproj]
C:\kevinransom\fsharp\vsintegration\src\FSharp.VS.FSI\fsiSessionToolWindow.fs(537,5): error FS3118: Incomplete value or function defini
tion. If this is in an expression, the body of the expression must be indented to the same column as the 'let' keyword. [C:\kevinransom
\fsharp\vsintegration\src\FSharp.VS.FSI\FSharp.VS.FSI.fsproj]
````

I will open an issue since it's clearly an error.  But .ToString is staying for now.

Reply:
Interesting...